homeassistant/components/tuya/climate.py

```diff
@@ -30,6 +30,7 @@
 TUYA_HVAC_TO_HA = {
     "auto": HVACMode.HEAT_COOL,
     "cold": HVACMode.COOL,
+    "eco": HVACMode.HEAT_COOL,
```

Comment:
This change is invalid, `eco` should be mapped to a preset, not an HVAC mode.

Reply:
Hi frenck,
Btw, I am very new in these implementation, sorry for that I have did it incorrectly.

Please take a look of the attached picture. You can see, that the eco is the one and only one option for the HVAC mode.
Because it is missing from the TUYA_HVAC_TO_HA, it can not be controlled from HA. The only available option is OFF.

![image](https://user-images.githubusercontent.com/8330252/197167199-cc0bd055-1e7a-4384-9bde-24341789ee9c.png)

Reply:
![image](https://user-images.githubusercontent.com/8330252/197167791-3221cb7d-69dd-4fbc-8e7b-c19df13b0d77.png)

Reply:
BTW. The newer model of these heater have more options. It has also the "eco" mode next to the "hot", "cold" and "auto", and they can be controlled from HA perfectly.
But I have three of the old models, and only two of the new ones.
Here is a screenshot of these models:
![image](https://user-images.githubusercontent.com/8330252/197168354-f9a9b55d-4da9-479b-9e91-6eb8bc14f238.png)

Reply:
> Please take a look of the attached picture. You can see, that the eco is the one and only one option for the HVAC mode.

I'm aware of the issue you are trying to solve.

But `eco` is not an HVAC mode in Home Assistant, it needs to be translated/mapped into preset in the Home Assistant world. Just adding it as a HVAC mode is incorrect and will lead to issues when overlapping modes are provided.